lib/AST/Type.cpp

```diff
@@ -2027,9 +2027,15 @@ static ForeignRepresentableKind
 getObjCObjectRepresentable(Type type, const DeclContext *dc) {
   // @objc metatypes are representable when their instance type is.
   if (auto metatype = type->getAs<AnyMetatypeType>()) {
+    auto instanceType = metatype->getInstanceType();
+
+    // Consider protocol metatypes as not Objective-C representable.
+    if (isa<MetatypeType>(metatype) &&
+        instanceType->getKind() == TypeKind::Protocol)
```

Comment:
Done.

Thanks for the tip! I've used `TypeBase::is` function instead, looks like it's more suitable here, please, correct me if I'm wrong.